controllers/testclustertopology_controller.go

```diff
@@ -698,6 +703,104 @@ func (r *TestClusterTopologyReconciler) initTiUPResources(ctx context.Context, c
 	return requeue, nil
 }
 
+func (r *TestClusterTopologyReconciler) initTiDBResources(ctx context.Context, ct *naglfarv1.TestClusterTopology, resources []*naglfarv1.TestResource, clusterIPMaps map[string]string) (bool, error) {
+	var requeue bool
+	var haProxyResources []*naglfarv1.TestResource
+	var tiupResources []*naglfarv1.TestResource
+	if ct.Spec.TiDBCluster.HAProxy != nil {
+		for i := 0; i < len(resources); i++ {
+			if resources[i].Name == ct.Spec.TiDBCluster.HAProxy.Host {
+				haProxyResources = append(haProxyResources, resources[i])
+			} else {
+				tiupResources = append(tiupResources, resources[i])
+			}
+		}
+	}
+	if len(tiupResources) == 0 {
```

Comment:
How about the user only fills TiDBCluster.HAProxy doesn't fill TiDB, TiKV, PD, etc components? You set HAProxy.Host to tiup resources?

Reply:
yes, add check

Reply:
done